src/actions/mod.rs

```diff
@@ -35,6 +35,21 @@ use std::time::Duration;
 
 use self::compiler_message_parsing::{FileDiagnostic, ParseError, Suggestion};
 
+// TODO: Support non-`file` URI schemes in VFS. We're currently ignoring them because
+// we don't want to crash the RLS in case a client opens a file under different URI scheme
+// like with git:/ or perforce:/ (Probably even http:/? We currently don't support remote schemes).
+macro_rules! ignore_non_file_uri {
```

Comment:
Actually `convert_pos_to_span` requires `doc` only to [retrieve](https://github.com/rust-lang-nursery/rls/pull/430/files#diff-53ac0aa2d451315631bac6f85bcfb7bdR770) a file `Path` and uses only that. Maybe we could change the function signature to accept `Path` instead, revert return type so it returns `Span` as before (so no error handling would be needed there) and then move the `parse_file_path` into the `ignore_non_file_uri` macro as suggested?